src/GitTfs.VsCommon/TfsHelper.Vs2017Base.cs

```diff
@@ -195,9 +196,14 @@ public override IIdentity GetIdentity(string username)
 
         protected override TfsTeamProjectCollection GetTfsCredential(Uri uri)
         {
-            var vssCred = HasCredentials
-                ? new VssClientCredentials(new WindowsCredential(GetCredential()))
-                : VssClientCredentials.LoadCachedCredentials(uri, false, CredentialPromptType.PromptIfNeeded);
+            VssCredentials vssCred;
+
+            if (HasPAT)
+                vssCred = new VssBasicCredential(string.Empty, PAT);
+            else if (HasUserName && HasPassword)
+                vssCred = new VssClientCredentials(new WindowsCredential(GetCredential()));
+            else
+                vssCred = new VssClientCredentials(new WindowsCredential(GetCredential()), CredentialPromptType.PromptIfNeeded);
```

Comment:
It could use an empty credential, but I though I would improve it a bit so that if they did pass username it would already be in the credential and they would only get prompted for username. So they can leave off username, and it works the old way, and or pass it in and only get prompted for the password. It doesn't break the old functionality, just extends it.

Reply:
Ok, I did some more testing, and it turns out there is a difference in the way this is handled for on premise versus Azure. I will bulletproof the code for both scenarios and update the docs.